lib/services/v2.0/kc2.dart

```diff
@@ -0,0 +1,817 @@
+    if (address == null) {
+      return null;
+    }
+
+    return ss58.Codec(42).encode(address.cast<int>());
+  }
+
+  void _processNewUserTransaction(
+      String hash,
+      BigInt timeStamp,
+      String accountId,
+      String signer,
+      String method,
+      String pallet,
+      String blockNumber,
+      Map<String, dynamic> args,
+      int blockIndex,
+      MapEntry<String, Object?>? failedReason,
+      Map<String, dynamic> rawData,
+      List<KC2Event> txEvents) async {
+    if (newUserCallback == null) {
+      return;
+    }
+
+    final username = args['username'];
+    final phoneNumberHash = hex.encode(args['phone_number_hash'].cast<int>());
+
+    final newUserTx = KC2NewUserTransactionV1(
+      accountId: accountId,
+      username: username,
+      phoneNumberHash: phoneNumberHash,
+      args: args,
+      pallet: pallet,
+      signer: signer,
+      method: method,
+      failedReason: failedReason,
+      timestamp: timeStamp,
+      hash: hash,
+      blockNumber: blockNumber,
+      blockIndex: blockIndex,
+      transactionEvents: txEvents,
+      rawData: rawData,
+    );
+
+    await newUserCallback!(newUserTx);
+  }
+
+  void _processUpdateUserTransaction(
+      String hash,
+      BigInt timeStamp,
+      String address,
+      String signer,
+      Map<String, dynamic> args,
+      MapEntry<String, Object?>? failedReason,
+      String method,
+      String pallet,
+      String blockNumber,
+      int blockIndex,
+      Map<String, dynamic> rawData,
+      List<KC2Event> txEvents) async {
+    final username = args['username'].value;
+    final phoneNumberHashOption = args['phone_number_hash'].value;
+    final phoneNumberHash = phoneNumberHashOption == null
+        ? null
+        : hex.encode(phoneNumberHashOption.cast<int>());
+
+    final updateUserTx = KC2UpdateUserTxV1(
+      username: username,
+      phoneNumberHash: phoneNumberHash,
+      args: args,
+      pallet: pallet,
+      signer: signer,
+      method: method,
+      failedReason: failedReason,
+      timestamp: timeStamp,
+      hash: hash,
+      blockNumber: blockNumber,
+      blockIndex: blockIndex,
+      transactionEvents: txEvents,
+      rawData: rawData,
+    );
+
+    await updateUserCallback!(updateUserTx);
+  }
+
+  void _processAppreciationTransaction(
+      String hash,
+      BigInt timeStamp,
+      String address,
+      String signer,
+      Map<String, dynamic> args,
+      MapEntry<String, Object?>? failedReason,
+      String method,
+      String pallet,
+      String blockNumber,
+      int blockIndex,
+      Map<String, dynamic> rawData,
+      List<KC2Event> txEvents) async {
+    debugPrint("Args: $args");
+
+    final to = args['to'];
+
+    // todo: tripple checks this - it comes from num
+    // @Danylo Kyrieiev
+
+    final BigInt amount = args['amount'];
+    final int communityId = args['community_id'].value;
+    final int charTraitId = args['char_trait_id'].value;
```

Comment:
I think this values can be null when users use appreciation as simple transfer

Reply:
Got it.